### PR TITLE
🌱 add workflow to approve actions on PR change

### PR DESCRIPTION
**Pull request by tuminoid**
If external contributor adds a PR, and a member /ok-to-test it, GH actions do not run until approved by an approver. This approval needs to be given if the PR content changes, so it inconveniences the contributor but also the approver.

This PR approves workflows, if ok-to-test label is present.

Adapted from CAPI workflow:
https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/pr-gh-workflow-approve.yaml

### CODE REVIEW

**Comment by tuminoid** [comment]
Not sure how to test it before merging, but given that it is from CAPI and its working for them, I'd say it should work the same for us.

We can test it here in IPAM first, then take it into CAPM3 and BMO as well (+standalone operator too)

**Comment by tuminoid** [comment]
/cc @lentzi90 @kashifest @elfosardo @dtantsur 
Would you be OK with this?

**Review by Sunnatillo — COMMENTED** [review]
LGTM

**Review by kashifest — COMMENTED** [review]
/approve

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Comment by tuminoid** [comment]
/hold
To get more opinions. I don't expect objections tho.

**Review by dtantsur — APPROVED** [review]

**Comment by tuminoid** [comment]
/override test-centos-e2e-integration-main  test-ubuntu-integration-main

**Comment by tuminoid** [comment]
Tested in PR https://github.com/metal3-io/ip-address-manager/pull/486

Before `/ok-to-test`:

image

After `/ok-to-test`:

image

NOTE, the initial approve needs to be given by the admin. It will not be given by the workflow. The workflow only re-approves once approved workflow, if /ok-to-test is present when PR is updated. This is what we want, so test passed.

**Comment by tuminoid** [comment]
Updated the explanation with case 1 and case 2 just to be super clear what this allows.

**Pull request merged**
